packages/common/src/api/README.md

```diff
@@ -0,0 +1,150 @@
+
+1.  Implement the fetch function
+
+    - `audiusClient` and `audiusBackend` are available from the context argument
+
+    ```typescript
+    endpoints: {
+        getSomeData: {
+            fetch: async (
+                    { id } /* fetch args */,
+                    { apiClient, audiusBackend } /* context */
+                ) => {
+                    return await apiClient.getSomeData({ id })
+                },
+            options: {
+                // see below
+            }
+        }
+    }
+    ```
+
+1.  Endpoint options
+
+    - **`schemaKey`** - the corresponding key in `apiResponseSchema` see [schema.ts](./schema.ts).
+
+      _Note: A schema key is required, though any unreserved key can be used if the data does not contain any of the entities stored in the entity cache (i.e. any of the `Kinds` from [Kind.ts](/packages/common/src/models/Kind.ts))_
+
+    - **`kind`** - in combination with either `idArgKey` or `permalinkArgKey`, allows cache hits from the entity cache prior to the first remote fetch
+      - **`idArgKey`** - must match a key the argument from the fetch function's `fetchArgs` containing the entity's id
+      - **`permalinkArgKey`** - must match a key the argument from the fetch function's `fetchArgs` containing the entity's permalink
+
+1.  Export hooks
+
+    A Hooks will automatically be generated for each endpoint, using the naming convention `` [`use${capitalize(endpointName)}`] `` (e.g. `getSomeData` -> `useGetSomeData`)
+
+    ```typescript
+    const userApi = createApi({
+      endpoints: {
+        getSomeData: {
+          // ...
+        }
+      }
+    })
+
+    // Export the hook for each endpoint here
+    export const { useGetSomeData } = userApi.hooks
+    export default userApi.reducer
+    ```
+
+## Calling the endpoint
+
+1.  Generated fetch hooks take the same args as the fetch function plus an options object. They return the same type returned by the fetch function.
+
+    ```typescript
+    type QueryHook = (
+        fetchArgs: /* matches the first argument to the endpoint fetch fn */
+        options: {
+            disabled: boolean
+        }
+    ) => {
+        data: /* return value from fetch function */
+        status: Status
+        errorMessage?: string
+    }
+    ```
+
+1.  In your component
+
+    ```typescript
+    const {
+      data: someData,
+      status,
+      errorMessage
+    } = useGetSomeData(
+      { id: elementId },
+      /* optional */ { disabled: !elementId }
+    )
+
+    return status === Status.LOADING ? (
+      <Loading />
+    ) : (
+      <DisplayComponent data={someData} />
+    )
+    ```
+
+## Debugging
+
+- [createApi.ts](./createApi.ts) contains the implementation of the fetch hooks. You can put breakpoints in `useQuery`. Tip: conditional breakpoints are especially useful since the core logic is shared across every audius-query hook. Try `endpoinName === 'myEndpoint && fetchArgs === { ...myArgs }'` to scope down to only your own hook
```

Comment:
```suggestion
- [createApi.ts](./createApi.ts) contains the implementation of the fetch hooks. You can put breakpoints in `useQuery`. Tip: conditional breakpoints are especially useful since the core logic is shared across every audius-query hook. Try `endpointName === 'myEndpoint && fetchArgs === { ...myArgs }'` to scope down to only your own hook
```